### PR TITLE
Regwall: Switches to GSI's "popup" mode (without iframe shim for now)

### PR DESCRIPTION
**Pull request by ChrisAntaki**
This PR will help me verify the flow E2E in the GNews app

### CODE REVIEW

File: examples/sample-pub/metering.js [hunk 1]
```diff
@@ -44,6 +44,17 @@ const MeteringDemo = {
     GaaMeteringRegwall.signOut().then(() => void location.reload());
   },
 
+  /** Mocks registration of a user, given data from Google Sign-In. */
```
Comment (elijahsoria) [inline examples/sample-pub/metering.js:47]: Could you add a comment about what pubs should do in production?

File: src/utils/gaa.js [hunk 2]
```diff
@@ -187,19 +187,35 @@ const REGWALL_HTML = `
...
+ *   getBasicProfile: function(): {
+ *     getName: function(): string,
+ *     getGivenName: function(): string,
+ *     getFamilyName: function(): string,
+ *     getImageUrl: function(): string,
+ *     getEmail: function(): string,
+ *   },
+ * }} GoogleSignInUserDef
+ */
+let GoogleSignInUserDef;
```
Comment (elijahsoria) [inline src/utils/gaa.js:204]: nit: Maybe rename to GoogleUserBasicProfile (or similar) to match the name used on the Google developer site.

File: src/utils/gaa.js [hunk 3]
```diff
@@ -251,11 +266,7 @@ export class GaaMeteringRegwall {
         .then(
           () =>
             // Only initialize Google Sign-In once.
-            self.gapi.auth2.getAuthInstance() ||
-            self.gapi.auth2.init({
-              'ux_mode': 'redirect',
-              'redirect_uri': redirectUri,
-            })
+            self.gapi.auth2.getAuthInstance() || self.gapi.auth2.init()
```
Comment (elijahsoria) [inline src/utils/gaa.js:269]: From the dev docs, it seems that client_id is a required parameter to init. Are you using something else?

https://developers.google.com/identity/sign-in/web/reference#gapiauth2initparams

File: src/utils/gaa.js [hunk 4]
```diff
@@ -231,10 +247,9 @@ export class GaaMeteringRegwall {
   /**
    * Configures Google Sign-In.
    * @nocollapse
-   * @param {{ redirectUri: string }=} params
    * @return {!Promise}
    */
-  static configureGoogleSignIn({redirectUri} = {redirectUri: ''}) {
+  static configureGoogleSignIn() {
```
Comment (elijahsoria) [inline src/utils/gaa.js:252]: Should this be made private? I see that it is getting called within renderGoogleSignInButton_(). Should users be able to call this function or only allow show()?

File: src/utils/gaa.js (hunk 2 above)
Comment (ChrisAntaki) [reply src/utils/gaa.js:204]: I updated this to GoogleUser, to match the docs. The basic profile is something the GoogleUser contains

File: examples/sample-pub/metering.js (hunk 1 above)
Comment (ChrisAntaki) [reply examples/sample-pub/metering.js:47]: I updated the docs to suggest setting a 1p http-only cookie so it lives past 7 days in Safari

File: src/utils/gaa.js (hunk 3 above)
Comment (ChrisAntaki) [reply src/utils/gaa.js:269]: I'm defining the Client ID in a meta tag. This is the simple method GSI tells developers about first in their docs https://developers.google.com/identity/sign-in/web/sign-in#specify_your_apps_client_id

**Review by elijahsoria — APPROVED** [review]

File: src/utils/gaa.js (hunk 4 above)
Comment (ChrisAntaki) [reply src/utils/gaa.js]: I just privatized it

**Review by elijahsoria — APPROVED** [review]

**Pull request merged**